app/controllers/concerns/foreman/controller/discovered_extensions.rb

```diff
@@ -70,19 +70,23 @@ def perform_reboot_all hosts = Host::Discovered.all
     result = true
     error_message = _("Errors during reboot: %s")
     overall_errors = ""
+    host_details  = []
 
     if hosts.count > 0
       hosts.each do |discovered_host|
         begin
           unless discovered_host.reboot
             error =  "#{discovered_host.name}: failed to reboot "
             overall_errors << error
-            logger.error error
+            host_details << {:name => "#{discovered_host.name}", :error => "#{error}"}
           end
         rescue Exception => e
-          error = "#{discovered_host.name}: #{e.to_s} "
+          error = "#{e.to_s} "
+          if error.include?"[Foreman::WrappedException]:" and error.include?"([Errno"
+            error = error[/]: (.*?) \(\[/, 1]
+          end
```

Comment:
@tstrachota alone `error = error[/]: (.*?) \(\[/, 1]` could have solved our problem of making the error message more readable but I thought it would be good to have the condition as well. The output of the command now looks like:
![screenshot from 2018-03-12 19-05-33](https://user-images.githubusercontent.com/10976840/37286622-6243781e-25fa-11e8-8d85-96428e9235a5.png)

Reply:
@tstrachota on thought though, do we really need the `message` node?,  we show the output in hammer-cli-foreman-disocvery using the `host_details`.

Reply:
I don't think we need the main `message` node in this case. We can keep it there just for the sake of consistency, but I'm also ok with leaving it out.

Reply:
@lzap @tstrachota Ohkay, keeping consistency in mind I have made changes to the pr and the tests are passing locally.

Reply:
It's important to keep it for backward compatibility, people have scripts hooked onto this V2 API, we don't change API until V3, so glad you kept it.

Reply:
Hmmm can't you do just:

```ruby
msg = if e.respons_to?(:wrapped_exception)
  e.wrapped_exception.message
else
 e.message
end
```

Or similar? This parsing is really bad, I'd rather fix how exceptions are being throwed rather than doing this magic.

Reply:
@lzap the code snippet that you have provided is really helpful. thanks!